app/src/main/java/org/oppia/app/hintsandsolution/HintsAndSolutionDialogFragment.kt

```diff
@@ -10,12 +10,18 @@ import org.oppia.app.fragment.InjectableDialogFragment
 import org.oppia.app.model.State
 import javax.inject.Inject
 
-private const val CURRENT_EXPANDED_LIST_INDEX_SAVED_KEY = "CURRENT_EXPANDED_LIST_INDEX"
-private const val STATE_SAVED_KEY = "STATE_SAVED_KEY"
-private const val HINT_INDEX_SAVED_KEY = "HINT_INDEX_SAVED_KEY"
-private const val IS_HINT_REVEALED_SAVED_KEY = "IS_HINT_REVEALED_SAVED_KEY"
-private const val SOLUTION_INDEX_SAVED_KEY = "SOLUTION_INDEX_SAVED_KEY"
-private const val IS_SOLUTION_REVEALED_SAVED_KEY = "IS_SOLUTION_REVEALED_SAVED_KEY"
+private const val CURRENT_EXPANDED_LIST_INDEX_SAVED_KEY =
+  "HintsAndSolutionDialogFragment.current_expanded_list_index"
+private const val STATE_SAVED_KEY =
+  "HintsAndSolutionDialogFragment.state"
+private const val HINT_INDEX_SAVED_KEY =
+  "HintsAndSolutionDialogFragment.hint_index"
+private const val IS_HINT_REVEALED_SAVED_KEY =
+  "HintsAndSolutionDialogFragment.is_hint_revealed"
+private const val SOLUTION_INDEX_SAVED_KEY =
+  "HintsAndSolutionDialogFragment.solution_index"
+private const val IS_SOLUTION_REVEALED_SAVED_KEY =
```

Comment:
Shift all these private keys to `companion object`, similar to what I mentioned for `AdministratorControlsActivity`.

Reply:
done
But I have used private keys only inside companion object also, so please check if I need to make those keys internal or not.